options/options.go

```diff
@@ -275,51 +242,8 @@ func ProcessConfig(apps ...string) error {
 	if err != nil {
 		return err
 	}
-
-	if !appOptions {
-		appsOptions, err := snapctl.Get("apps").Run()
-		if err != nil {
-			return err
-		}
-		globalOptions, err := snapctl.Get("config").Run()
-		if err != nil {
-			return err
-		}
-		if isSet(appsOptions) || isSet(globalOptions) {
-			var migratable string
-			if env.SnapName == "edgexfoundry" {
-				migratable = `
-Exception: The following internally set env options are automatically migrated:
-	- env.security-secret-store.add-secretstore-tokens
-	- env.security-secret-store.add-known-secrets
-	- env.security-bootstrapper.add-registry-acl-roles
-Note: Disabling app-options WILL NOT revert the migration!`
-			}
-			return fmt.Errorf("app options (prefix `apps.' or 'config.') are allowed only when app-options is true.\n\n%s%s",
-				"WARNING: Setting app-options=true WILL UNSET existing env options and ignore future sets!!",
-				migratable)
-
-		} else if isSet(envOptions) {
-			// return and continue with legacy option handling
-			return nil
-		} else {
-			// no app options or env options are set
-			// continue to cleanup previously set env vars from files
-			log.Debug("No app options are set.")
-		}
-	}
-
 	if isSet(envOptions) {
-		if err := migrateLegacyInternalOptions(); err != nil {
-			return err
-		}
-
-		// It is important to unset any options to avoid conflicts in
-		// 	deprecated configure hook processing
-		if err := snapctl.Unset("env").Run(); err != nil {
-			return err
-		}
-		log.Info("Unset all 'env.' options.")
+		return fmt.Errorf("disallowed env options: %s", envOptions)
 	}
```

Comment:
Why does not remove this function? env option is not supported in v3 anyway.

Reply:
I remove it and some other useless code: https://github.com/canonical/edgex-snap-hooks/pull/78/commits/13bf0f6cc50640d3159e7f92f9b0812a84b9c06d